application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml

```diff
@@ -59,13 +61,19 @@
       #set($list = $obj.getProperty('flash').getValue())
       #if($list == $util.getNull())
         #set($discard = $obj.set('flash', [$xcontext.user]))
-        #set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
       #elseif($list.class.name == 'com.xpn.xwiki.objects.ListProperty$NotifyList' &amp;&amp; !$list.contains($xcontext.user))
         #set($discard = $list.add($xcontext.user))
         #set($discard = $obj.set('flash', $list))
-        #set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
       #end
     #end
+    ## Check if the message should be marked as seen today.
+    #set ($msgObj = $msgdoc.getObject('Flash.FlashValidationClass'))
+    #set ($seenDate = $msgObj.getValue('seenDate'))
+    #set ($wasSeenToday = $datetool.format($dateFormat, $dateNow) == $datetool.format($dateFormat, $seenDate))
```

Comment:
I assume you want to compare the dates by ignoring the time component, but this depends on the date format, which you don't control if it comes from the wiki preferences. What happens if an administrator adds the time to the date format? If the date format is used only for this comparison then you should have full control over it.

Reply:
So it is safer to use a string for `$datetool.format`

Reply:
Yes, I think so, it would be more clear that the reason you format the date is because you want to "round" the date (ignore the time component).